documentation/docs/03-loading.md

```diff
@@ -4,6 +4,10 @@ title: Loading
 
 A component that defines a page or a layout can export a `load` function that runs before the component is created. This function runs both during server-side rendering and in the client, and allows you to get data for a page without (for example) showing a loading spinner and fetching data in `onMount`.
 
+> Since `load` can run on the server as well as the client browser, any state changes to a variable that happens on the server will be saved and persisted, and all clients hitting the server-rendered page will be exposed to the already mutated state instead of a clean one.
+>
+> It is recommended to make any state changes in the `<script>` block or outside of `load`. To share states exclusive to individual clients, use [`request.locals`](#hooks-handle) or [session](#loading-input-session).
```

Comment:
> It is recommended to make any state changes in the `<script>` block or outside of `load`.

I don't know that I'd go so far as to make this a recommendation

Reply:
I was thinking on how to _not_ recommend making state changes in `load` function, but I think that it's better to use a positive connotation and recommend something instead of discouraging without giving alternatives. Any thoughts how we should go about it?

Reply:
I don't think there's any issue with making state changes in `load` though. It seems much too broad. You should just be aware if you mutate a singleton that it will persist across calls

Reply:
Isn't the issue is exactly that, can you explain why it's too broad? I'm not aware of any other stuff on the pages that can also run on the server other than `load`.

> You should just be aware if you mutate a singleton that it will persist across calls

I think the users are aware of this, but the underlying problem is that they expect it to be persisted across calls in the client's side and not from the server directly, and really the difference is just mutating in `load` and passing it to `<script>` first before mutating, to make sure that singleton is always mutated in the client's browser.

Reply:
A state change is just mutating some variable. It seems too broad to say you can't do that

I was under the impression that the problem discussed in the issue (https://github.com/sveltejs/kit/issues/1542) was only talking about the server-side behavior. They were surprised that the counter was incrementing on the server between different requests and users

Reply:
It's not really saying they can't, but I guess it does sounds like there's implicit restrictions and may cause more confusion. Removed in the latest change.